lib/rpm_package.py

```diff
@@ -15,16 +16,103 @@
 
 import os
 import logging
+import re
 import yaml
 
+import rpmUtils
+
 from lib import config
 from lib import exception
+from lib import utils
 from lib.package import Package
 
 CONF = config.get_config().CONF
 LOG = logging.getLogger(__name__)
 
 
+def compare_versions(v1, v2):
+    return rpmUtils.miscutils.compareEVR((None, v1, None), (None, v2, None))
+
+
+class SpecFile(object):
+
+    def __init__(self, path):
+        self.path = path
+        self._content = None
+        self._cached_tags = dict()
+
+    @property
+    def content(self):
+        """
+        Get and cache the content of the spec file.
+        """
+        if self._content is None:
+            with open(self.path, 'r+') as file_:
+                self._content = file_.read()
+        return self._content
+
+    @content.setter
+    def content(self, value):
+        """
+        Set the cached content of the spec file.
+        """
+        self._content = value
+
+    def write_content(self):
```

Comment:
Ok, but we might then introduce a bug when using rpmdev-bumpspec which also writes to the spec file. Calling `bump_release` in between calls to `write_content` would lose the bump information.